test/test_nn.py

```diff
@@ -3910,6 +3910,40 @@ def test_Conv2d_inconsistent_types_on_GPU_without_cudnn(self):
             # but it should work with the same type
             nn.functional.conv2d(inputs.float(), weights.float(), bias.float())
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    def test_Conv2d_size_1_kernel_on_GPU_without_cudnn(self, dtype=torch.float):
+        inputs = torch.randn(2, 3, 5, 5, device="cuda", dtype=dtype, requires_grad=True)
+        with cudnn.flags(enabled=True, benchmark=True, deterministic=True):
+            conv1 = torch.nn.Conv2d(3, 3, kernel_size=1).to("cuda", dtype)
+            conv2 = torch.nn.Conv2d(3, 3, kernel_size=1).to("cuda", dtype)
+            conv2.bias.data.copy_(conv1.bias.data)
+            conv2.weight.data.copy_(conv1.weight.data)
+            out1 = conv1(inputs)
+            out2 = conv2(inputs)
+            self.assertEqual(out1, out2, atol=0.0, rtol=0)
+            y = torch.randn(out1.size(), device="cuda", dtype=dtype)
+            out1.backward(y)
+            out2.backward(y)
+            self.assertEqual(conv1.bias.grad.data, conv2.bias.grad.data, atol=0.0, rtol=0)
+            self.assertEqual(conv1.weight.grad.data, conv2.weight.grad.data, atol=0.0, rtol=0)
```

Comment:
It seems like you are comparing that 2 exactly same convolutions produce same result, but what if this result is wrong?

Reply:
I was copying one of the existing check (I guess I should compare cuda vs cudnn results within a certain delta?)

Reply:
Or you could compare with cpu.